knowhere/index/vector_index/helpers/RangeUtil.h

```diff
@@ -0,0 +1,165 @@
+// Copyright (C) 2019-2020 Zilliz. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance
+// with the License. You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software distributed under the License
+// is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+// or implied. See the License for the specific language governing permissions and limitations under the License.
+
+#pragma once
+
+#include <vector>
+
+#include <faiss/impl/AuxIndexStructures.h>
+#include <faiss/MetricType.h>
+
+#include "knowhere/common/Exception.h"
+#include "knowhere/common/Log.h"
+#include "knowhere/index/vector_index/helpers/IndexParameter.h"
+#include "knowhere/utils/BitsetView.h"
+
+namespace knowhere {
+
+inline bool
+distance_in_range(const float dist, const float low_bound, const float high_bound, const bool is_ip) {
+    return ((is_ip && low_bound < dist && dist <= high_bound) || (!is_ip && low_bound <= dist && dist < high_bound));
+}
+
+// filter range search results with distance falling in range between low_bound and high_bound
+inline void
+GetRangeSearchResultForOneNq(const int64_t i_size,
+                             const float* i_distances,
+                             const int64_t* i_labels,
+                             const bool is_ip,
+                             const float low_bound,
+                             const float high_bound,
+                             const int64_t o_size,
+                             float* o_distances,
+                             int64_t* o_labels,
+                             const faiss::BitsetView bitset) {
+    size_t num = 0;
+    for (size_t i = 0; i < i_size; i++) {
+        auto dis = i_distances[i];
+        auto id = i_labels[i];
+        KNOWHERE_THROW_IF_NOT_MSG(bitset.empty() || !bitset.test(id), "bitset invalid");
+        KNOWHERE_THROW_IF_NOT_FMT((is_ip && dis > low_bound) || (!is_ip && dis < high_bound),
+                                  "distance %f invalid, is_ip %s, low_bound %f, high_bound %f",
+                                  dis, (is_ip ? "true" : "false"), low_bound, high_bound);
+        if (distance_in_range(dis, low_bound, high_bound, is_ip)) {
+            o_labels[num] = id;
+            o_distances[num] = dis;
+            num++;
+        }
+    }
+    KNOWHERE_THROW_IF_NOT_FMT(num == o_size, "%ld not equal %ld", num, o_size);
+}
+
+inline void
+GetRangeSearchResult(const faiss::RangeSearchResult& res,
```

Comment:
two `GetRangeSearchResult` without slightly discrepancy on signature without any comment is poor in readability. Instead of simply gathering all the common codes as a function, it will be more readable and maintainable to create function according to their unit functionality.
In this case, better solution is:
1. Create a function only works on filtering our of range data.
2. Create another function to assemble data.
3. If no use case for filtering single nq, we don't have to make it an individual function to increase the stack call.

Reply:
updated